src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp

```diff
@@ -109,142 +138,59 @@ class boss_magmadar : public CreatureScript
     }
 };
 
-// Serrated Bites timer may be wrong
-class npc_magmadar_core_hound : public CreatureScript
+// 19411 Lava Bomb
+// 20474 Lava Bomb
+class spell_magmadar_lava_bomb : public SpellScriptLoader
 {
 public:
-    npc_magmadar_core_hound() : CreatureScript("npc_magmadar_core_hound") { }
-
-    struct npc_magmadar_core_houndAI : public CreatureAI
+    spell_magmadar_lava_bomb() : SpellScriptLoader("spell_magmadar_lava_bomb")
     {
-        npc_magmadar_core_houndAI(Creature* creature) : CreatureAI(creature)
-        {
-        }
-
-        EventMap events;
-        std::list<Creature*> hounds;
-        bool smoldering = false;
-        Unit* killer;
-
-        void removeFeignDeath()
-        {
-            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PREVENT_EMOTES_FROM_CHAT_TEXT);
-            me->RemoveFlag(UNIT_FIELD_FLAGS_2, UNIT_FLAG2_FEIGN_DEATH);
-            me->RemoveFlag(UNIT_DYNAMIC_FLAGS, UNIT_DYNFLAG_DEAD);
-            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE);
-            me->RemoveUnitMovementFlag(MOVEMENTFLAG_ROOT);
-            me->ClearUnitState(UNIT_STATE_DIED);
-            me->ClearUnitState(UNIT_STATE_CANNOT_AUTOATTACK);
-            me->DisableRotate(false);
-        }
-
-        void DamageTaken(Unit* attacker, uint32& damage, DamageEffectType /*damagetype*/, SpellSchoolMask /*damageSchoolMask*/) override
-        {
-            if (me->HealthBelowPctDamaged(0, damage))
-            {
-                if (!smoldering)
-                {
-                    killer = attacker;
-                    events.ScheduleEvent(EVENT_IGNITE, 10000);
-                    me->SetHealth(1);
-                    me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PREVENT_EMOTES_FROM_CHAT_TEXT);
-                    me->SetFlag(UNIT_FIELD_FLAGS_2, UNIT_FLAG2_FEIGN_DEATH);
-                    me->SetFlag(UNIT_DYNAMIC_FLAGS, UNIT_DYNFLAG_DEAD);
-                    me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE);
-                    me->AddUnitMovementFlag(MOVEMENTFLAG_ROOT);
-                    me->AddUnitState(UNIT_STATE_DIED);
-                    me->AddUnitState(UNIT_STATE_CANNOT_AUTOATTACK);
-                    me->DisableRotate(true);
-                    Talk(EMOTE_SMOLDERING);
-                }
-                damage = 0;
-                smoldering = true;
-            }
-        }
-
-        void Reset() override
-        {
-            removeFeignDeath();
-        }
-
-        void JustDied(Unit* /*killer*/) override
-        {
-            removeFeignDeath();
-        }
+    }
 
-        void EnterCombat(Unit* /*victim*/) override
-        {
-            events.ScheduleEvent(EVENT_SERRATED_BITE, 10000); // timer may be wrong
-        }
+    class spell_magmadar_lava_bomb_SpellScript : public SpellScript
```

Comment:
You need to validatr the spells used inside a spell script.
So in this case SPELL_LAVA_BOMB_EFFECT and

Reply:
Like now?